onnxruntime/test/ir/onnx_model_test.cc

```diff
@@ -162,5 +162,16 @@ TEST(ONNXModelsTest, TestIRv4NonInputInitializers) {
   ASSERT_TRUE(Model::Load("testdata/subgraph_implicit_input_from_initializer.onnx", model).IsOK());
   EXPECT_TRUE(model->MainGraph().Resolve().IsOK());
 }
+
+// test a model that has an op with a FunctionBody and one of the nodes within the FunctionBody has a subgraph in it.
+// The test model has is an opset-11 op with a 'Range' node.
+// 'Range' has a FunctionBody and has a 'Loop' node with a subgraph.
+// Graph::Resolve to succeed when processing the subgraph pertaining to the overall FunctionBody.
+TEST(ONNXModelsTest, TestIRv4NonInputInitializers) {
+  std::shared_ptr<Model> model;
+  ASSERT_TRUE(Model::Load("testdata/model_containing_op_with_function_body.onnx", model).IsOK());
+  EXPECT_TRUE(model->MainGraph().Resolve().IsOK());
+}
```

Comment:
nit: it's nicer to output the status if this fails
  auto status = model->MainGraph().Resolve();
  EXPECT_TRUE(status.IsOK()) << status;

We probably should have a EXPECT_STATUS_OK macro that does that though for convenience

Reply:
Fixed.